### DISCUSSION TITLE
Unable to get it to work

### DISCUSSION THREAD

**Discussion by devbyVeriTech** [pockethost/pockethost#299] [Q&A]
Thanks for the project, but I can't get it to work.
You said we just have change the ('http://127.0.0.1:8090') to ('http://nextflixapp.pockethost.io')......

Now the thing is, I can create users and signin users using the ('http://127.0.0.1:8090') but it doesnt work with the ('http://nextflixapp.pockethost.io').

I also checked my (nextflixapp.pockethost.io/_/#/logs) and  there is no log there.

what do you think the problem is???
I'm using SvelteKit

**Comment by benallfree** [comment]
Hi @devbyVeriTech, it sounds like your code is not even hitting PocketHost. Please paste a screen shot of the Chrome network debugger and show that a successful connection is being made.

You can also find us on Discord if that is easier.

**Reply by devbyVeriTech** [reply]
![codee](https://github.com/pockethost/pockethost/assets/131156642/63e36bc8-ec0a-42c5-9eb6-1a495517654b)

This is the code for my +page.server.ts and it still doesn't work.
but if i replace that ('https://nextflixapp.pockethost.io') with ('http://127.0.0.1'), it'll work properly


import PocketBase from 'pocketbase'
import { redirect } from "@sveltejs/kit";
const pb = new PocketBase('https://nextflixapp.pockethost.io');


export const actions = {
    register: async ({request}) => {
        const body = Object.fromEntries(await request.formData())

        try {
            await pb.collection('users').create(body);
        } catch (e){
            console.error(e);
            throw e
        }

        throw redirect (303, '/login');
    }
}

**Reply by benallfree** [reply]
Can you please show your Chrome debugger network area? Also, you might have better luck on our Discord :)

**Reply by devbyVeriTech** [reply]
And again, when I use ('https://nextflixapp.pockethost.io/') on +page.svelte as a function and bind it with the on:submit event, it works perfectly. It just doesn't work on +page.server.ts file

**Reply by benallfree** [reply]
Are you trying to use the PocketBase JS SDK from the server-side? Have you followed the instructions for that?

**Reply by devbyVeriTech** [reply]
I've been using it with the local host pocketbase and it works well.
Few minutes ago, it worked well with your instance and after restarting my VScode, it stopped working
I used your instance for authentication and it worked well.
Now, everything stopped

**Reply by benallfree** [reply]
Try making a simple nodejs app that does a call to PocketHost and see if that works

**Reply by devbyVeriTech** [reply]
I'll try that and get back to you

**Reply by devbyVeriTech** [reply]
So sorry, but it still didn't work.
Since you're familiar with Sveltekit and server-side rendering, maybe you should try it yourself and see

**Reply by benallfree** [reply]
Sure, send me a repo link and I'll try it :)

**Reply by devbyVeriTech** [reply]
https://github.com/devbyVeriTech/new

**Reply by benallfree** [reply]
This is what I get:

```bash
11:42:04 AM [vite-plugin-svelte] /Volumes/Code/repos/pockethost/pockethost/forks/new/src/routes/+page.svelte:4:15 Page has unused export property 'data'. If it is for external reference only, please consider using `export const data`
11:42:04 AM [vite-plugin-svelte] /Volumes/Code/repos/pockethost/pockethost/forks/new/src/routes/+page.svelte:4:15 Page has unused export property 'data'. If it is for external reference only, please consider using `export const data`
ClientResponseError 400: Failed to create record.
    at file:///Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/pocketbase/dist/pocketbase.es.mjs:1:34060
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async sign (/Volumes/Code/repos/pockethost/pockethost/forks/new/src/routes/+page.server.ts:12:7)
    at async Module.handle_action_json_request (/Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/@sveltejs/kit/src/runtime/server/page/actions.js:57:16)
    at async resolve (/Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/@sveltejs/kit/src/runtime/server/respond.js:415:18)
    at async Module.respond (/Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/@sveltejs/kit/src/runtime/server/respond.js:282:20)
    at async file:///Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:510:22 {
  url: 'https://nextflixapp.pockethost.io/api/collections/users/records',
  status: 400,
  response: {
    code: 400,
    message: 'Failed to create record.',
    data: { password: [Object] }
  },
  isAbort: false,
  originalError: {
    url: 'https://nextflixapp.pockethost.io/api/collections/users/records',
    status: 400,
    data: { code: 400, message: 'Failed to create record.', data: [Object] }
  }
}
ClientResponseError 400: Failed to create record.
    at file:///Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/pocketbase/dist/pocketbase.es.mjs:1:34060
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async sign (/Volumes/Code/repos/pockethost/pockethost/forks/new/src/routes/+page.server.ts:12:13)
    at async Module.handle_action_json_request (/Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/@sveltejs/kit/src/runtime/server/page/actions.js:48:16)
    at async resolve (/Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/@sveltejs/kit/src/runtime/server/respond.js:407:18)
    at async Module.respond (/Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/@sveltejs/kit/src/runtime/server/respond.js:274:20)
    at async file:///Volumes/Code/repos/pockethost/pockethost/forks/new/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:510:22 {
  url: 'https://nextflixapp.pockethost.io/api/collections/users/records',
  status: 400,
  response: {
    code: 400,
    message: 'Failed to create record.',
    data: { password: [Object] }
  },
  isAbort: false,
  originalError: {
    url: 'https://nextflixapp.pockethost.io/api/collections/users/records',
    status: 400,
    data: { code: 400, message: 'Failed to create record.', data: [Object] }
  }
}
```

**Reply by benallfree** [reply]
`Failed to create record` means it is connecting to your PocketBase instance, but the request is failing due to PocketBase security. Do you get something else?

**Reply by devbyVeriTech** [reply]
Yeah
I get something else.
The url doesn't show 'https://nextflixapp.pockethost.io/api/collections/users/records', it shows ' ', an empty space

**Reply by devbyVeriTech** [reply]
ClientResponseError 0: Something went wrong while processing your request.
    at file:///C:/Users/BEATBOSS/Documents/Tuts/gamer/node_modules/pocketbase/dist/pocketbase.es.mjs:1:34060
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async sign (C:\Users\BEATBOSS\Documents\Tuts\gamer\src\routes\+page.server.ts:12:13)
    at async Module.handle_action_json_request (C:\Users\BEATBOSS\Documents\Tuts\gamer\node_modules\@sveltejs\kit\src\runtime\server\page\actions.js:48:16)
    at async resolve (C:\Users\BEATBOSS\Documents\Tuts\gamer\node_modules\@sveltejs\kit\src\runtime\server\respond.js:407:18)
    at async Module.respond (C:\Users\BEATBOSS\Documents\Tuts\gamer\node_modules\@sveltejs\kit\src\runtime\server\respond.js:274:20)
    at async file:///C:/Users/BEATBOSS/Documents/Tuts/gamer/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:510:22 {
  url: '',
  status: 0,
  response: {},
  isAbort: false,
  originalError: TypeError: fetch failed
      at fetch (C:\Users\BEATBOSS\Documents\Tuts\gamer\node_modules\undici\index.js:110:15)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async sign (C:/Users/BEATBOSS/Documents/Tuts/gamer/src/routes/+page.server.ts:12:7)
      at async Module.handle_action_json_request (C:/Users/BEATBOSS/Documents/Tuts/gamer/node_modules/@sveltejs/kit/src/runtime/server/page/actions.js:57:16)
      at async resolve (C:/Users/BEATBOSS/Documents/Tuts/gamer/node_modules/@sveltejs/kit/src/runtime/server/respond.js:415:18)
      at async Module.respond (C:/Users/BEATBOSS/Documents/Tuts/gamer/node_modules/@sveltejs/kit/src/runtime/server/respond.js:282:20)
      at async file:///C:/Users/BEATBOSS/Documents/Tuts/gamer/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:510:22 {
    cause: Error: connect EACCES 104.21.8.253:443
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1532:16) {
      errno: -4092,
      code: 'EACCES',
      syscall: 'connect',
      address: '104.21.8.253',
      port: 443
    }
  }
}

**Reply by benallfree** [reply]
What version of node are you running?

**Reply by devbyVeriTech** [reply]
19.8.1

**Reply by benallfree** [reply]
That should be okay. This seems like an issue with your local machine though.

Here's what ChatGTP had to say:

------------------

Yes, I recognize this. It's an error typically encountered by Node.js developers.

Here's a breakdown of the error:

- **EACCES**: This error code stands for "Error: Access." It means that the process did not have the necessary permissions to perform a specific operation.

- **connect**: The operation that caused the error was an attempt to "connect" to a remote server or service.

- **104.21.8.253:443**: This is the IP address and port to which the application tried to connect. The port 443 is typically used for HTTPS connections.

- **TCPConnectWrap.afterConnect**: This is a callback function from the Node.js core `net` module, which gets invoked after attempting a TCP connection. The error occurred after this connect attempt.

In simpler terms, your Node.js application tried to connect to a server at IP `104.21.8.253` on port `443` and encountered a permissions issue.

Some potential causes and solutions for this error:

1. **Insufficient Permissions**: Ensure that the Node.js process has the necessary permissions to make network connections.
   
2. **Firewall or Security Software**: It's possible that a firewall or security software is preventing the connection. Check if any such software is blocking outgoing connections on port 443.

3. **Binding on Low Ports**: Sometimes, binding to low ports (below 1024) requires elevated permissions. However, since you're connecting to port 443 and not binding to it, this is less likely to be the issue.

4. **Run as a Different User**: If you're running your Node.js app as a user with restricted permissions, try running it as a different user.

5. **Check the Code**: Ensure that there's no code logic that's restricting the connection or the permissions.

6. **External Restrictions**: Sometimes, the external server or network infrastructure might be causing the issue. For example, some cloud providers restrict certain types of outgoing connections.

To troubleshoot further, you may want to:
- Check system logs for any additional information.
- Use debugging tools or increase the logging level of your Node.js app.
- Test the connection outside of your Node.js app (using tools like `curl` or `telnet`) to see if it's an application-specific issue or a system-wide issue.

**Reply by devbyVeriTech** [reply]
Aiit, I'll get back to you soon

**Reply by devbyVeriTech** [reply]
But does that mean, I'll face the same issue if I deploy it now?
Or I'll have to debug my PC first?

**Reply by benallfree** [reply]
I think you should try a different machine (or deploying it) to see if you get the same issue. I cannot reproduce your issue on my machine.

**Reply by devbyVeriTech** [reply]
Aiit but please, I'll be happy if you can retry it just for confirmation because like I told you some days back which you might actually forget, sometimes it works and most times it doesn't

**Reply by devbyVeriTech** [reply]
I deployed it and it worked well.
Thank you so much
